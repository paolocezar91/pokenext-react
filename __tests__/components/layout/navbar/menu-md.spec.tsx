import MenuMd from "@/components/layout/navbar/menu-md";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

jest.mock("@/components/layout/navbar/search", () =>
  jest.fn(() => <div>Search</div>),
);
jest.mock("@/components/layout/navbar/nav-link", () =>
  jest.fn(({ href, children }) => <a href={href}>{children}</a>),
);
jest.mock("@/components/layout/navbar/nav-user-auth/nav-user-auth", () =>
  jest.fn(() => <div>nav-user-auth</div>),
);
jest.mock("@/context/user-context", () => ({
  useUser: () => ({
    settings: {},
  }),
}));

describe("MenuMd", () => {
  it("renders children inside the menu", () => {
    render(<MenuMd></MenuMd>);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<MenuMd></MenuMd>);
    expect(asFragment()).toMatchSnapshot();
  });
});
