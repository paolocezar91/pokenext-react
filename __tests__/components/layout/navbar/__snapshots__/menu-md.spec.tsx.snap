// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuMd matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="hidden w-full md:block md:w-auto"
  >
    <ul
      class="   flex   flex-col   items-center   p-4   mt-4   border   border-gray-700   rounded   rtl:space-x-reverse   md:p-0   md:space-x-4   md:flex-row   md:mt-0   md:border-0   md:bg-(--pokedex-red)   "
    >
      <li
        class="h-10"
      >
        <div
          class="go-to rounded"
        >
          <div>
            Search
          </div>
        </div>
      </li>
      <li
        class="h-10"
      >
        <div>
          nav-user-auth
        </div>
      </li>
      <li />
    </ul>
  </div>
</DocumentFragment>
`;
